Inlining trivial method to suppress static analyzer warnings in PageLoadState

basuke opened the pull request:
#### 4b5fd9d5f7c7ab98481c47360abac045a88d0490
<pre>
Inlining trivial method to suppress static analyzer warnings in PageLoadState
<a href="https://bugs.webkit.org/show_bug.cgi?id=280340">https://bugs.webkit.org/show_bug.cgi?id=280340</a>
<a href="https://rdar.apple.com/problem/137133185">rdar://problem/137133185</a>

Reviewed by Ben Nham.

Because PageLoadState is now ref/deref object, it should be clear that the method is trivial or not.

* Source/WebKit/UIProcess/PageLoadState.cpp:
(WebKit::PageLoadState::isLoading const): Deleted.
(WebKit::PageLoadState::hasUncommittedLoad const): Deleted.
(WebKit::PageLoadState::activeURL const): Deleted.
(WebKit::PageLoadState::wasPrivateRelayed const): Deleted.
* Source/WebKit/UIProcess/PageLoadState.h:
(WebKit::PageLoadState::isLoading const):
(WebKit::PageLoadState::hasUncommittedLoad const):
(WebKit::PageLoadState::activeURL const):
(WebKit::PageLoadState::wasPrivateRelayed const):

Canonical link: <a href="https://commits.webkit.org/285557@main">https://commits.webkit.org/285557@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/acc183f11a376bbec4cd96920922a930038e943e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73010 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52439 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25818 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77217 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24248 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/60244 "Built successfully") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57383 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15869 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76077 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47398 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62839 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37801 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44038 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20309 "Build is in progress. Recent messages:") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22577 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65897 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20664 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78887 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-2-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65831 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62841 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65115 "Found 1 new API test failure: /TestWebKit:WebKit.InjectedBundleInitializationUserDataCallbackWins (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16097 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7093 "Passed tests") | | 
| | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-18-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/3026 "Built successfully") | | 
| | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-11-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-11-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/PageLoadState.h:
```diff
@@ -142,24 +142,24 @@ class PageLoadState {
     void reset(const Transaction::Token&);
 
-    bool isLoading() const;
+    inline bool isLoading() const;
```
You don't need `inline` here and the others below. Having `inline` on the definition is sufficient, no need to have it on the declaration as well.

cdumez commented on Source/WebKit/UIProcess/PageLoadState.h:
```diff
@@ -269,4 +269,65 @@ class PageLoadState {
+    return isLoading(m_uncommittedState);
+}
+
+inline bool PageLoadState::isLoading(const Data& data)
```
I'll let @rniwa and @geoffreygaren comment but it doesn't seem *that* small, not sure we want to inline this.

basuke replied:
I agree. I think it's already trivial access without inlining and I hope analyzer can detect this well.

geoffreygaren replied:
To clarify, the analyzer can only detect that a function is trivial if the function is inline. So, we have to choose between
* Make a function inline, so we can avoid RefPtr
* Use RefPtr, so we can keep a function out-of-line

Given that this inline doesn't require new #includes, maybe inline is OK?

Note that 'inline' does not mandate emitting the body of the function into the caller, and non-'inline' does not mandate the opposite, either. The compiler uses its own judgment. It can out-of-line a function marked 'inline', or inline a function not marked 'inline' (if its definition is visible in the translation unit, which it often is in our unified builds).

cdumez replied:
> Given that this inline doesn't require new #includes, maybe inline is OK?

If it doesn't bloat the binary / hurt perf.

basuke committed:
Inlining trivial method to suppress static analyzer warnings in PageLoadState
https://bugs.webkit.org/show_bug.cgi?id=280340
rdar://problem/137133185

Reviewed by Ben Nham.

Because PageLoadState is now ref/deref object, it should be clear that the method is trivial or not.

* Source/WebKit/UIProcess/PageLoadState.cpp:
(WebKit::PageLoadState::isLoading const): Deleted.
(WebKit::PageLoadState::hasUncommittedLoad const): Deleted.
(WebKit::PageLoadState::activeURL const): Deleted.
(WebKit::PageLoadState::wasPrivateRelayed const): Deleted.
* Source/WebKit/UIProcess/PageLoadState.h:
(WebKit::PageLoadState::isLoading const):
(WebKit::PageLoadState::hasUncommittedLoad const):
(WebKit::PageLoadState::activeURL const):
(WebKit::PageLoadState::wasPrivateRelayed const):

Canonical link: https://commits.webkit.org/285557@main
